src/main/java/edu/harvard/iq/dataverse/engine/command/impl/UningestFileCommand.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.DataFile;
+import edu.harvard.iq.dataverse.DatasetVersion;
+import edu.harvard.iq.dataverse.FileMetadata;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.dataaccess.DataAccess;
+import edu.harvard.iq.dataverse.dataaccess.StorageIO;
+import edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.PermissionException;
+import edu.harvard.iq.dataverse.util.FileUtil;
+import edu.harvard.iq.dataverse.util.StringUtil;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.logging.Logger;
+import javax.persistence.Query;
+
+/**
+ *
+ * @author skraffmi
+ */
+@RequiredPermissions({})
+public class UningestFileCommand extends AbstractVoidCommand  {
+
+    private static final Logger logger = Logger.getLogger(MoveDatasetCommand.class.getCanonicalName());
+    final DataFile uningest;
+    
+    public UningestFileCommand(DataverseRequest aRequest, DataFile uningest) {
+        super(aRequest, uningest);
+        this.uningest = uningest;
+    }
+
+    @Override
+    protected void executeImpl(CommandContext ctxt) throws CommandException {
+        
+        // first check if  user is a superuser
+        if ( (!(getUser() instanceof AuthenticatedUser) || !getUser().isSuperuser() ) ) {      
+            throw new PermissionException("Uningest File can only be called by Superusers.",
+                this,  Collections.singleton(Permission.EditDataset), uningest);                
+        }
+        
+        String fileName;
+
+            StorageIO<DataFile> dataAccess = null;
+            Long originalFileSize;
+
+            try{
+                dataAccess = DataAccess.getStorageIO(uningest);
+                dataAccess.open();
+                originalFileSize = dataAccess.getAuxObjectSize(FileUtil.SAVED_ORIGINAL_FILENAME_EXTENSION);
+                fileName = dataAccess.getFileName();
+                uningest.setFilesize(originalFileSize);
+                dataAccess.revertBackupAsAux(FileUtil.SAVED_ORIGINAL_FILENAME_EXTENSION);
+            }
+        
+        
+            catch (IOException ioex) {
```

Comment:
If we catch an exception here, we really need to bail here, instead of proceeding with the rest of the uningest. Continuing with a warning here means that if, for example, we failed to replaced the tab file with the saved original, we'll end up with a file that has the Stata extension and mime type, but is still a tabular file.